# Stats assignment 2

In [3]:
import numpy as np

### 1. In each of the following situations, state whether it is a correctly stated hypothesis testing problem and why?

### We know that in hypothesisi testing we have a null hypothesis and an alternate hypothesis...
### The alternate hypothesis should be the opposite of alternate hypothesis
### The hypothesis test are only for population values

1. True condition satisfied
2. False 𝐻0: 𝜎 > 10 but 𝐻1: 𝜎 = 10 which is not exact opposite like 𝜎 < 10
3. False  condition not satisfied its about sample so...
4. False  both seem to be stating different p equals different values... the hypothesis are not opposite to each other
5. False  the alternate hypothesis is not opposite to null hypothesis... s=30 so alternate should be s != 30 and they are about samples

### 2. The college bookstore tells prospective students that the average cost of its textbooks is Rs. 52 with a standard deviation of Rs. 4.50. A group of smart statistics students thinks that the average cost is higher. To test the bookstore’s claim against their alternative, the students will select a random sample of size 100. Assume that the mean from their random sample is Rs. 52.80. Perform a hypothesis test at the 5% level of significance and state your decision.

### 3. A certain chemical pollutant in the Genesee River has been constant for several years with mean μ = 34 ppm (parts per million) and standard deviation σ = 8 ppm. A group of factory representatives whose companies discharge liquids into the river is now claiming that they have lowered the average with improved filtration devices. A group of environmentalists will test to see if this is true at the 1% level of significance. Assume \ that their sample of size 50 gives a mean of 32.5 ppm. Perform a hypothesis test at the 1% level of significance and state your decision.

In [7]:
mew = 34
std_dev = 8
n = 50
x_bar = 32.5
alpha = 0.01

z = (x_bar - mew)/std_dev*np.sqrt(n)
z

-1.3258252147247767

### 4.Based on population figures and other general information on the U.S. population, suppose it has been estimated that, on average, a family of four in the U.S. spends about 1135 dollars annually on dental expenditures. Suppose further that a regional dental association wants to test to determine if this figure is accurate for their area of country. To test this, 22 families of 4 are randomly selected from the population in that area of the country and a log is kept of the family’s dental expenditure for one year. The resulting data are given below. Assuming, that dental expenditure is normally distributed in the population, use the data and an alpha of 0.5 to test the dental association’s hypothesis.


In [13]:
mew = 1135
n=22
alpha = 0.5

data = [1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699,
872, 913, 944, 954, 987, 1695, 995, 1003, 994]

x_mean = np.mean(data)
print(x_mean)
std_dev = np.std(data)
print(std_dev)

1031.3181818181818
234.84799462314425


In [12]:
std_err = std_dev/np.sqrt(n)

t_score = (x_mean - mew)/std_err
print(t_score)

-2.070747228595759


### 5.In a report prepared by the Economic Research Department of a major bank the Department manager maintains that the average annual family income on Metropolis is 48,432 dollars. What do you conclude about the validity of the report if a random sample of 400 families shows and average income of 48,574 dollars with a standard deviation of 2000?

In [18]:
# compute z test statistics at 0.1 percent significance

mew = 48432
std_dev = 2000
n=400
x_bar = 48574
alpha = 0.01

z = (x_bar - mew)/std_dev*np.sqrt(n)
z

1.42

### 6.  Suppose that in past years the average price per square foot for warehouses in the United States has been 32.28 dollars. A national real estate investor wants to determine whether that figure has changed now. The investor hires a researcher who randomly samples 19 warehouses that are for sale across the United States and finds that the mean price per square foot is 31.67 dollars, with a standard deviation of 1.29 dollars. assume that the prices of warehouse footage are normally distributed in population. If the researcher uses a 5% level of significance, what statistical conclusion can be reached? What are the hypotheses?

In [23]:
h0 : mew = 32.28
h1 : mew != 32.28

In [24]:
mew = 32.28
n=19
alpha = 0.05

x_mean = 31.67
std_dev = 1.29

In [25]:
std_err = std_dev/np.sqrt(n)

t_score = (x_mean - mew)/std_err
print(t_score)

-2.06118477175179


### 7. Fill in the blank spaces in the table and draw your conclusions from it.

| acceptance region | sample size | alpha | beta & mew = 52 | beta & mew = 50.5 |
|--- | --- |--- | --- |--- | 
| 48.5 < x_mean < 51.5 | 10 |  | |
| 48 < x_mean < 52 | 10 |  | |
| 48.81 < x_mean < 51.9 | 16 |  | |
| 48.42 < x_mean < 51.58 | 16 |  | |



### 8.  Find the t-score for a sample size of 16 taken from a population with mean 10 whenthe sample mean is 12 and the sample standard deviation is 1.5.

In [27]:
mew =  10
n=16

x_mean = 12
std_dev = 1.5

std_err = std_dev/np.sqrt(n)

t_score = (x_mean - mew)/std_err
print(t_score)

5.333333333333333


### 9. Find the t-score below which we can expect 99% of sample means will fall if samples of size 16 are taken from a normally distributed population

### 10. If a random sample of size 25 drawn from a normal population gives a mean of 60 and a standard deviation of 4, find the range of t-scores where we can expect to find the middle 95% of all sample means.

### 11. Two-tailed test for difference between two population means Is there evidence to conclude that the number of people travelling from Bangalore to Chennai is different from the number of people travelling from Bangalore to Hosur in a week, given the following:

In [60]:
# H0: there is no difference
# H1 : there is a difference

# Population 1: Bangalore to Chennai 
n1 = 1200
X1 = 452
S1 = 212
# Population 2: Bangalore to Hosur 
n2 = 800
X2 = 523
S2 = 185

# n1 = 1000  
# n2 = 2000
# X1 = 76.5
# X2= 77
# S1=S2= 7


df =  n1 + n2 -2
SE = np.sqrt(((np.power(S1,2)/n1) + (np.power(S2,2)/n2)))
# var = (∑[X-X(mean)]^2 + ∑[Y-Y(mean)]^2))/(n1 + n2 -2)


Z_test = (X1- X2)/ SE
Z_test

-7.926428526759299

Since this is for 2 tail and we take confidence interval of 95%, we get...

Z(2.5%) = 1.96  and Z(-2.5%) = -1.96

Z(-2.5) > Z_test < Z(2.5)

The z score lies in the rejection region so, we reject the null hypothesis

### 12. Is there evidence to conclude that the number of people preferring Duracell battery is different from the number of people preferring Energizer battery, given the following:

In [59]:
# H0: there is no difference
# H1 : there is a difference

# Population 1: Duracell
n1 = 100
X1 = 308
S1 = 84
# Population 2: Energizer
n2 = 100
X2 = 254
S2 = 67

SE = np.sqrt(((np.power(S1,2)/n1) + (np.power(S2,2)/n2)))
Z_test = (X1- X2)/ SE
Z_test

5.025702668336442

Since this is for 2 tail and we take confidence interval of 95%, we get...

Z(2.5%) = 1.96  and Z(-2.5%) = -1.96

Z(-2.5) < Z_test > Z(2.5)

The z score lies in the rejection region so, we reject the null hypothesis

### 13. Pooled estimate of the population variance Does the data provide sufficient evidence to conclude that average percentage increase in the price of sugar differs when it is sold at two different prices?

In [63]:
# Population 1: Price of sugar = Rs. 27.50 
n1 = 14
X1 = 0.317 
S1 = 0.12
# Population 2: Price of sugar = Rs. 20.00 
n2 = 9
X2 = 0.21
S2 = 0.11

SE = np.sqrt(((np.power(S1,2)/n1) + (np.power(S2,2)/n2)))
Z_test = (X1- X2)/ SE
Z_test

2.1965113720074068

Since this is for 2 tail and we take confidence interval of 95%, we get...

Z(2.5%) = 1.96  and Z(-2.5%) = -1.96

Z(-2.5) < Z_test > Z(2.5)

The z score lies in the rejection region so, we reject the null hypothesis

### 14. The manufacturers of compact disk players want to test whether a small price reduction is enough to increase sales of their product. Is there evidence that the small price reduction is enough to increase sales of compact disk players?

In [66]:
# H0: the sales are not affected
# H1: The sales are increasing
    
# Population 1: Before reduction
n1 = 15
X1 = 6598 
S1 = 844
# Population 2: After reduction 
n2 = 12
X2 = 6870
S2 = 669

SE = np.sqrt(((np.power(S1,2)/n1) + (np.power(S2,2)/n2)))
Z_test = (X1- X2)/ SE
Z_test

-0.9341299576730813

Since this is for right tail and we take confidence interval of 95%, we get...

Z(2.5%) = 1.96  and Z(-2.5%) = -1.96

Z(-2.5) < Z_test > Z(2.5)

The z score lies in the rejection region so, we reject the null hypothesis

### 15. Comparisons of two population proportions when the hypothesized difference is zero  Carry out a two-tailed test of the equality of banks’ share of the car loan market in 1980 and 1995.

#H0: the shares in car loan market hasn't changed
#H1: the shares in car loan market hasn changed

#Population 1: 1980
n1 = 1000
X1 = 53
S1 = 0.53
#Population 2: 1985
n2 = 100
X2 = 43
S2= 0.53

df = n1 + n2 -2 = 1098

for 2 tailed t test with df = 1098 and 0.1% significance we get T-Value (two-tailed): +/- 2.580314

our p value is greater then our t value so we accept our null hypothesis

### 16. Carry out a one-tailed test to determine whether the population proportion of traveler’s check buyers who buy at least 2500 dollars in checks when sweepstakes prizes are offered as at least 10% higher than the proportion of such buyers when no sweepstakes are on.


### 17. A die is thrown 132 times with the following results: 

In [77]:
Number_turned_up = np.array([1, 2, 3, 4, 5, 6])
Frequency = np.array([16, 20, 25, 14, 29, 28])
# Is the die unbiased? Consider the degrees of freedom as 𝑝 − 1

In [90]:
# (H0): The die is unbiased
# (HA): The die is not unbiased.
    
    
# we will perform a chi-squared test
    
# On the hypothesis that the die is unbiased we should expect the frequency of each number to be 

E = 132/6 # = 22
n=6
X = np.sum(np.square(Number_turned_up - E))/E
print(X)

alpha = 0.05

df = 5   # n-1

critical_value = 11.0705
print(critical_value)

# Since X > critical_value


# HO is rejected.
# H1 is accepted

94.13636363636364
11.0705


### 18. In a certain town, there are about one million eligible voters. A simple random sample of 10,000 eligible voters was chosen to study the relationship between gender and participation in the last election. The results are summarized in the following 2X2 (read two by two) contingency table:

|   | Men | Women |
|---|---|---|
|voted |2792 | 3591|
|notvoted |1486 |2131 |

### We would want to check whether being a man or a woman (columns) is independent ofhaving voted in the last election (rows). In other words, is “gender and voting independent”?


#H0: sex is independent of voting
#H1: sex is depedent of voting
    
    
we will perform hi squared test.. for which we need the observed table with totals and expected table


OBSERVED TABLE

||Men|Women|Total|
|---|---|---|---|
|Voted|2792|3591  |6383|
|Didn't vote|1486|2131  |3617|
Total|4278|5722  |10000|

chance of (row_i,col_j) = (chance row_i) * (chance col_j)

 (row_i,col_j) = N*(chance row_i)*(chance col_j) = (Sum row_i)*(Sum col_j) / N

EXPECTED TABLE
		  
||Men|Women |Total|
|---|---|---|---|
|Voted		|2731	|3652  |	6383|
|Didn't vote|	1547|	2070  |	3617|
Total	|	4278	|5722  |	10000

In [94]:
c11 = (2792-2731)**2/2731
c12 = (3591-3652)**2/3652
c21 = (1486-1547)**2/1547
c22 = (2131-2070)**2/2070

X = c11 + c12 + c21 + c22
print("chi value")
print(X)

chi value
6.584283456757529


#Now we compute the P-value. 
#This is done by looking under the Chi-square table with (rows-1)*(cols-1) degrees of freedom. 
#In the case of a 2x2 table (our case) the number of degrees of freedom is (2-1)(2-1)=1*1=1. 
#The table gives the tail areas at:
    
|Degrees of freedom|99%|10%|5%|1%|
|---|---|---|---|---|
|1|0.00016|2.71|3.84|6.64|
|2 |0.020|4.60|5.99|9.21|

Since the observed X2 = 6.58 and thus,
3.84 < X2 < 6.64
we conclude that:
1% < P-value < 5%
and we reject the NULL. The data supports the hypothesis that sex and voting are dependent in this town.

### 19. A sample of 100 voters are asked which of four candidates they would vote for in an election. The number supporting each candidate is given below:

|higgins | reardon |white |charlton|
|---|---|---|---|
|41 |19 | 24 | 16 |

### Do the data suggest that all candidates are equally popular? [Chi-Square = 14.96, with 3 df, 𝑝 0.05 .

 A Chi-Squared Goodness-of-Fit test is appropriate here. The null hypothesis is that there is no preference for any of the candidates: if this is so, we would expect roughly equal numbers of voters to support each candidate. Our expected frequencies are therefore 100/4 = 25 per candidate.

|O |41| 19| 24| 16| 
|---|---|---|---|---|
|E |25 |25 |25 |25 |
|(O-E) |16 |-6 |-1| -9 |
|(O-E)squared |256| 36 |1| 81| 
|(O-E)squared/E| 10.24 |1.44 |0.04 |3.24|

Adding together the last row gives us our value of X  :

X = 10.24+ 1.44 + 0.04 + 3.24 = 3.24 with 3 degrees of freedom

The critical value of Chi-Square for a 0.05 significance level and 3 d.f. is 7.82. Our obtained Chi-Square value is bigger than this, and so we conclude that our obtained value is unlikely to have occurred merely by chance. In fact, our obtained value is bigger than the critical Chi-Square value for the 0.01 significance level (13.28). In other words, it is possible that our obtained Chi-Square value is due merely to chance, but highly unlikely: a Chi-Square value as large as ours will occur by chance only about once in a hundred trials. It seems more reasonable to conclude that our results are not de to chance, and that the data do indeed suggest that voters do not prefer the four candidates equally.

 

### 20. Children of three ages are asked to indicate their preference for three photographs of adults. Do the data suggest that there is a significant relationship between age and photograph preference? What is wrong with this study? [Chi-Square = 29.6, with 4 df: 𝑝 < 0.05].

(a) Work out the row, column and grand totals (as shown in the shaded parts of the table, above).
(b) Work out the expected frequencies, using the formula:

|O| 18|22|20|2|28|40|20|10|40|
|---|---|---|---|---|---|---|---|---|---|
|E| 12|18|30|14|21|35|14|21|35|
|(O-E)| 6|4|-10|-12|7|5|6|11|5|
|(O-E)squared |36|16|100|144|49|25|36|121|25|
|(O-E)squared/E |3|0.89|3.33|10.29|2.33|0.71|2.57|5.76|0.71|
 
    
chi-squared is the sum of these:

c2 = 29.60.

d.f. = (rows - 1) * (columns - 1) = 2 * 2 = 4.

 
The critical value of Chi-Square in the table for a 0.001 significance level and 4 d.f. is 18.46. Our obtained Chi-Square value is bigger than this: therefore we have a Chi-Square value which is so large that it would occur by chance only about once in a thousand times. It seems more reasonable to accept the alternative hypothesis, that there is a significant relationship between age of child and photograph preference.

### 21. A study of conformity using the Asch paradigm involved two conditions: one where one confederate supported the true judgement and another where no confederate gave the correct response. Is there a significant difference between the "support" and "no support" conditions in the frequency with which individuals are likely to conform? [Chi-Square = 19.87, with 1 df: 𝑝 < 0.05].

 Here we have a 2x2 contingency table. Chi-Square is the appropriate test to use, but since we have 1 d.f., we will modify the formula to include "Yates' correction for continuity".

||support| no support | row totals |
|---|---|---|---|
|conform| 18|40|58|
|not conform| 32|10|42|
|column total| 50|50|100|
    
(a) Calculate the row, column and grand totals.

(b) Calculate the expected frequency for each cell of the table, by multiplying together the appropriate row and column totals and then dividing by the grand total.

(c) Subtract each expected frequency from its associated observed frequency; but then apply Yates' correction, by subtracting 0.5 from the absolute value of each O-E value. (The vertical bars in the formula mean "ignore any minus signs").

     
|O| 18|40|32|10|
|---|---|---|---|---|
|E| 29|29|21|21|
|(O-E 0.5)| 10.5| 10.5| 10.5| 10.5|
|(O-E 0.5)squared |110.25|110.25|110.25|110.25|
|(O-E)squared/E |3.80|3.80|5.25|5.25|

Chi-squared is the sum of these:

X2 = 18.10.

d.f. = (rows - 1) * (columns - 1) = 1 * 1 = 1.

Our obtained value of Chi-Squared is bigger than the critical value of Chi-Squared for a 0.001 significance level. In other words, there is less than a one in a thousand chance of obtaining a Chi-Square value as big as our obtained one, merely by chance. Therefore we can conclude that there is a significant difference between the "support" and "no support" conditions, in terms of the frequency with which individuals conformed.

### 22. We want to test whether short people differ with respect to their leadership qualities (Genghis Khan, Adolf Hitler and Napoleon were all stature-deprived, and how many midget MP's are there?) The following table shows the frequencies with which 43 short people and 52 tall people were categorized as "leaders", "followers" or as "unclassifiable". Is there a relationship between height and leadership qualities? [Chi-Square = 10.71, with 2 df: 𝑝 < 0.01].

 Expected frequencies are in brackets:
 
||short|tail|row totals|
|---|---|---|---|
|leader| 12(19.92) | 32(24.08) | 44 |
|follower| 22(16.29) | 14(19.71) | 36 |
|unclassifiable | 9(6.79) | 6(8.21) | 15 |
|column totals| 43|52|95|
        
 Chi-Square  = 3.146 + 2.602 + 1.998 + 1.652 + 0.720 + 0.595 = 10.712, with 2 d.f.
10.712 is bigger than the tabulated value of Chi-Square at the 0.01 significance level. We would conclude that there seems to be a relationship between height and leadership qualities. Note that we can only say that there is a relationship between our two variables, not that once causes the other. There could be all kinds of explanations for such a relationship.

### 23. Each respondent in the Current Population Survey of March 1993 was classified as employed, unemployed, or outside the labor force. The results for men in California age 35- 44 can be cross-tabulated by marital status, as follows: Men of different marital status seem to have different distributions of labor force status. Or is this just chance variation? (you may assume the table results from a simple random sample.)

H0: Rows and columns are independent.

H1: There is a relationship between rows and columns.
       (Significance Level = 0.05)


  Row   |     Col 1     Col 2     Col 3 |       Sum
  
  ------+------------------------+----------
  
  Obs 1 |       679       103       114 |    896.00
  
  Exp 1 |    654.06    109.29    132.65 |    896.00
  
     +------------------------+   
        
  Obs 2 |        63        10        20 |     93.00
  
  Exp 2 |     67.89     11.34     13.77 |     93.00
  
     +-------------------------+        
        
  Obs 3 |        42        18        25 |     85.00
  
  Exp 3 |     62.05     10.37     12.58 |     85.00
  
  ------+-------------------------+----------
  
  Sum   |    784.00    131.00    159.00 |   1074.00


  ChiSq = 
     (679 - 654.063315)^2 / 654.063315 + 
     (103 - 109.288641)^2 / 109.288641 + 
     (114 - 132.648045)^2 / 132.648045 + 
     (63 - 67.888268)^2 / 67.888268 + 
     (10 - 11.343575)^2 / 11.343575 + 
     (20 - 13.768156)^2 / 13.768156 + 
     (42 - 62.048417)^2 / 62.048417 + 
     (18 - 10.367784)^2 / 10.367784 + 
     (25 - 12.583799)^2 / 12.583799
    = 31.613103

    Degree of Freedom = (3 - 1)(3 - 1) = 4
    Critical Value: ChiSq(0.05, 4) = 9.487729

    ChiSq = 31.613103

    P(>ChiSq) = 2.295E-006 < 0.05

    Reject the null hypothesis at the 0.05 significance level.
    There is a relationship between rows and columns.
    